File: api/internal/server/root.go
package server

import (
	"encoding/json"
	"net/http"
)

func (s *Server) Root(w http.ResponseWriter, r *http.Request) {
	w.WriteHeader(http.StatusOK)
	json.NewEncoder(w).Encode(map[string]string{
		"message": "Hi! You probably meant to go to one of the other routes. Make sure to check the documentation!",
	})

}

File: apirust/src/routes/root.rs
use axum::{Json, response::IntoResponse};
use serde_json::json;

pub async fn root() -> Json<serde_json::Value> {
    Json(json!({
        "message":  "Hi! You probably meant to go to one of the other routes. Make sure to check the documentation!",
        "docs_url": "https://github.com/joswayski/creditcardhoroscope/blob/main/api/src/main.rs"
    }))
}
